src/concurrency/transaction_context.cpp

```diff
@@ -80,102 +78,63 @@ void TransactionContext::Init(const size_t thread_id,
 
   isolation_level_ = isolation;
 
-  is_written_ = false;
```

Comment:
That's effectively what @lmwnshn is doing in #1402.

Reply:
I think it is okay if we get rid of `insert_count_` but we should definitely use `is_written_`. It will help with multi statement transactions that are not explicitly set to read only.